boa/src/syntax/parser/expression/mod.rs

```diff
@@ -0,0 +1,875 @@
+//! Expression parsing.
+//!
+//! More information:
+//!  - [MDN documentation][mdn]
+//!  - [ECMAScript specification][spec]
+//!
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators
+//! [spec]: https://tc39.es/ecma262/#sec-ecmascript-language-expressions
+
+use super::{
+    read_formal_parameters, ArrowFunction, Block, Cursor, FunctionExpression, ParseError,
+    ParseResult, TokenParser,
+};
+use crate::syntax::ast::{
+    constant::Const,
+    keyword::Keyword,
+    node::{MethodDefinitionKind, Node, PropertyDefinition},
+    op::{AssignOp, BinOp, NumOp, UnaryOp},
+    punc::Punctuator,
+    token::TokenKind,
+};
+
+/// Generates an expression parser.
+///
+/// This macro has 2 mandatory identifiers:
+///  - The `$name` identifier will contain the name of the parsing structure.
+///  - The `$lower` identifier will contain the parser for lower level expressions.
+///
+/// Those exressions are divided by the punctuators passed as the third (and following) parameters.
+macro_rules! expression { ( $name:ident, $lower:ident, $( $op:path ),* ) => {
```

Comment:
why did you remove `[`...`]` form expression?

Reply:
I thought this made it easier to understand. What do you think? It's pretty easy to put it back, but I wanted to try how it looked like this.

Reply:
I see. I think separating the parameters of the `expression` macro is better. So its not a continuous line of parameters.
I would changed even more to be:
```rust
macro_rules! expression {
	(
	  name: $name:ident,
	  call: $lower:ident,
	  matches: [$( $op:path ),*]
	)
	...
}
```
This way it is clear to see that the name of the expression, what it will call and what it matches against.
What do you think? @Razican 
Am I overdoing it a bit?

Reply:
Hmmm the thing is that the `name` would be a `struct` or an `enum` to implement `TokenParser` in, so maybe a better name would suit. I will implement it this way for now, though, to see how it goes.